feat: 회원의 헬스장 관리 API 구현 (#24)

Lee-sh98 opened the pull request:
## 개요
- close #24 

## 작업사항

- **생성**
  - GymApiController
  - GymRequestDto
  - GymResponseDto
  - ListResponseDto

- **구현**
  - AuthorityRepository
    - deleteByUserIdAndGymId 메소드
    - findAuthoritiesByUserId 메소드
  - AuthorityService
    - register 메소드
    - delete 메소드
    - list 메소드

Lee-sh98 committed:
feat: 빌더패턴 생성자 추가 (#24)

Lee-sh98 committed:
feat:Authority 레포지토리 메소드 구현 (#24)

deleteByUserIdAndGymId
findAuthoritiesByUserId

Lee-sh98 committed:
feat: 리스트 dto 구현 (#24)

Lee-sh98 committed:
feat: GymResponseDto 구현 (#24)

Lee-sh98 committed:
feat: GymRequestDto 구현 (#24)

Lee-sh98 committed:
feat: GymApiController 구현 (#24)

Lee-sh98 committed:
feat: AuthorityService 구현 (#24)

Sangwook02 reviewed (commented):
comment 달아뒀습니다!
확인해보시고 논의해야 할 부분이 있어보이면 말씀해주세요!

Lee-sh98 committed:
feat: api 엔드포인트 수정, 함수명 변경 (#24)

change
/api/v1/gyms into
/api/v1/gyms/authority

change list into listRegistration

Lee-sh98 committed:
rename: controller 이름 변경 (#24)

엔드포인트 변경,
@Valid 적용,
GymListResponse 적용

Lee-sh98 committed:
rename: requestDto 이름 변경 (#24)

GymRequestDto.java ->
AuthorityRequestDto.java

Lee-sh98 committed:
feat: 함수명 변경, 반환형 변경 (#24)

함수명 변경
list() -> listRegistration()
반환형 변경
ListResponseDto -> GymListResponse

Lee-sh98 committed:
feat: GymListResponse 구현 (#24)

Lee-sh98 committed:
remove: ListResponseDto 삭제 (#24)

Sangwook02 reviewed (approved):
확인했습니다.

hoyun06 reviewed (approved):
Good
